cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,21 +1,41 @@
+import _ = require("cypress/types/lodash");
+import {MenuContentPage} from "../page/index";
+import {ProductsListPage} from "../page/index";
+import {ShopingCartPage} from "../page/index";
+import {LoginPage} from "../page/index";
+import {AddressStepPage} from "../page/index";
+import {ShippingStepPage} from "../page/index";
+import {PaymentStepPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shopingCartPage = new ShopingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
-  it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-
-    cy.get("#SubmitLogin").click();
-    cy.get(".cart_navigation button").click();
-    cy.get("#cgv").check();
-    cy.get("#form button").click();
-    cy.get("#HOOK_PAYMENT a.bankwire").click();
+  it("then should be bought a t-shirt", () => {
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.clickAddCartBtn();
+    productsListPage.clickProceedCheckoutBtn();
+
+    shopingCartPage.clickProceedCheckoutBtn();
+
+    loginPage.typeEmailUser();
+    loginPage.typePasswordUser();
+    loginPage.clickProceedSubmitLoginBtn();
+
+    addressStepPage.clickProceedCheckoutBtn();
+
+    shippingStepPage.checkTermsCheckbox();
+    shippingStepPage.clickProceedCheckoutBtn();
+
+    paymentStepPage.clickBankWireBtn();
+
     cy.get("#cart_navigation button").click();
 
     cy.get("#center_column > div > p > strong").should(
```

Comment:
the cy.get part also should be a method of the paymentStepPage object, so at the end you should have something like this:
```js
paymentStepPage.getConfirmationMessage().should(
    "have.text",
    "Your order on My Store is complete.",
);
```